ooni/common/txextra.py

```diff
@@ -191,12 +197,22 @@ def _handleRedirect(self, response, method, uri, headers, redirectCount):
             response.request.absoluteURI,
             locationHeaders[0]
         )
+        uri = client.URI.fromBytes(location)
+        if self.ignorePrivateRedirects and (is_private_address(uri.host) or
+                                            uri.host == "localhost"):
+            return response
```

Comment:
Yes this is a very valid point.

Thanks for bringing it up.

I will allow everything except loopback to pass through in this case.

> Also, it does not provide security as domain name like tplinkwifi.net can be used in Location as well.

Yes this is a limitation of the current approach, however in order to exclude also domain based redirects, given our current implementation of the HTTP test template, we would have to either pay the price of an extra A lookup before doing any HTTP request or we would have to heavily patch twisted to hook the domain resolution.

In light of the fact that we are going to be soon transition to using measurement-kit for network testing I think it makes most sense to defer this to future work and consider this as a poormans solution.

It may be worthwhile, though, to also look for redirects to stuff that is `.local`.